File: Apache_Simulator/Source/Apache_Simulator/Helicopter Logic/ApacheJoystick.cpp
// Fill out your copyright notice in the Description page of Project Settings.


#include "ApacheJoystick.h"
#include "B_Apache.h"
#include "Components/StaticMeshComponent.h"
#include "Components/ActorComponent.h"
#include "Math/UnrealMathUtility.h"
#include "Transform.h"
// Sets default values for this component's properties
UApacheJoystick::UApacheJoystick()
{
	// Set this component to be initialized when the game starts, and to be ticked every frame.  You can turn these features
	// off to improve performance if you don't need them.
	PrimaryComponentTick.bCanEverTick = true;

	// ...
}


// Called when the game starts
void UApacheJoystick::BeginPlay()
{
	Super::BeginPlay();
}

void UApacheJoystick::PitchUp()
{
	JoystickRotation.Y -= 10;
}
void UApacheJoystick::PitchDown()
{
	JoystickRotation.Y += 10;
}
void UApacheJoystick::YawRight()
{
	JoystickRotation.Z += 10;
}
void UApacheJoystick::YawLeft()
{
	JoystickRotation.Z -= 10;
}
void UApacheJoystick::RollRight()
{
	JoystickRotation.X -= 10;
}
void UApacheJoystick::RollLeft()
{
	JoystickRotation.X += 10;
}
// Called every frame
void UApacheJoystick::TickComponent(float DeltaTime, ELevelTick TickType, FActorComponentTickFunction* ThisTickFunction)
{
	Super::TickComponent(DeltaTime, TickType, ThisTickFunction);
	
	JoystickRotation.X = FMath::Clamp(JoystickRotation.X, -90.0f, 90.0f);
	JoystickRotation.Y = FMath::Clamp(JoystickRotation.Y, -90.0f, 90.0f);
	JoystickRotation.Z = FMath::Clamp(JoystickRotation.Z, -90.0f, 90.0f);

	ApacheRotation = JoystickRotation * Multiplier;
	
	Apache->Body->AddTorqueInDegrees(JoystickRotation );
	
//	JoystickRotation = FMath::Lerp(JoystickRotation, FVector(0,0,0), DeltaTime);

}

File: ApacheSim/Assets/Scripts/HelicopterLogic/ApacheJoystick.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class ApacheJoystick : MonoBehaviour
{


    public Vector3 JoystickRotation;


    public float Multiplier;


    public Vector3 ApacheRotation;


    public B_Apache Apache;

    // Start is called before the first frame update
    void Start()
    {
        
    }

    // Update is called once per frame
    void Update()
    {
        JoystickRotation.x = transform.rotation.eulerAngles.x + -24.176f;
        JoystickRotation.y = transform.rotation.eulerAngles.y + -78.679f;
        JoystickRotation.z = transform.rotation.eulerAngles.z + -26.051f;

        JoystickRotation.x = Mathf.Clamp(JoystickRotation.x, -90.0f, 90.0f);
        JoystickRotation.y = Mathf.Clamp(JoystickRotation.y, -90.0f, 90.0f);
        JoystickRotation.z = Mathf.Clamp(JoystickRotation.z, -90.0f, 90.0f);
        ApacheRotation = JoystickRotation * Multiplier;
        Apache.FlightPhysics.AddTorque(JoystickRotation);
    }



    void PitchUp()
    {
        JoystickRotation.y -= 10;
    }
    void PitchDown()
    {
        JoystickRotation.y += 10;
    }
    void YawRight()
    {
        JoystickRotation.z += 10;
    }
    void awLeft()
    {
        JoystickRotation.z -= 10;
    }
    void RollRight()
    {
        JoystickRotation.x -= 10;
    }
    void RollLeft()
    {
        JoystickRotation.x += 10;
    }
    // Called every frame



}
